bundles/org.openhab.ui/web/src/components/widgets/layout/oh-canvas-embedded-svg-mixin.js

```diff
@@ -0,0 +1,403 @@
+     * - useProxyElementForState: Use "flash" element to highlight the active state. The element is marked with the attribute flash: true and must be part of the elements group
+     * - useDisplayState: Use the formatted state value to write into tspan
+     * - stateOnColor: Color that should to be used when State is ON
+     * - stateOffColor: Color that should to be used when State is OFF
+     * - stateAsOpacity: Use the state from 0 to 100 as element opacity
+     * - stateMinOpacity: This allows an opacity to be kept above this value.
+     * - invertStateOpacity: 1 - opacity
+     * - stateOnAsStyleClass: Provide element-id:classname, separate multiple entries with comma. ON sets the class, if OFF is not provided, OFF removes the class of given element
+     * - stateOffAsStyleClass: Provide element-id:classname, separate multiple entries with comma. OFF sets the class
+     *
+     * @param {string} item item name that has triggered the state change
+     * @param {object} state object
+     * @param {object} svgElementConfig configuration of the svg element
+     * @param {string} svgElement the svg element that has been configured to represent the state
+     */
+    applyStateToSvgElement (item, stateObj, svgElementConfig, svgElement) {
+      const state = (svgElementConfig.useDisplayState) ? stateObj.displayState : stateObj.state
+      const stateType = stateObj.type
+      console.info(`Update ${svgElement.id} due to ${item} changing to ${state} ${stateType}`)
+      const tagName = svgElement.tagName
+      const stateOnColorRgbStyle = this.toRGBStyle(svgElementConfig.stateOnColor)
+      const stateOffColorRgbStyle = this.toRGBStyle(svgElementConfig.stateOffColor)
+
+      if (tagName === 'tspan') {
+        svgElement.innerHTML = state
+      }
+
+      switch (stateType) {
+        // currently no distinction is made regarding different state-types, only the following are supported (yet)
+        case 'OpenClosed':
+        case 'Percent':
+        case 'HSB':
+        case 'OnOff':
+          const useProxy = tagName === 'g' && svgElementConfig.useProxyElementForState // if proxy should be used and element is of type group
+          const element = (useProxy) ? svgElement.querySelector('[flash]') : svgElement
+          if (!element) {
+            console.warn(`Element ${svgElement} is a group element but has no containing element with the attribute "flash"`)
+            return
+          }
+          if (state === 'ON' || stateType === 'HSB') {
+            if (useProxy && svgElementConfig.stateAsOpacity) { // we use the flash element
+              let opacity = (state === 'ON') ? 1 : 0
+              opacity = (svgElementConfig.invertStateOpacity) ? 1 - opacity : opacity
+              opacity = (opacity < svgElementConfig.stateMinOpacity) ? svgElementConfig.stateMinOpacity : opacity
+              // TODO: use fill-opacity if fill not available
```

Comment:
What about this to do?

Reply:
This is meant for later improvement of SVG usage but I need to look further into this topic in which cases that might happen at all.

Reply:
Fine with me :+1: